library/src/main/java/pl/rspective/voucherify/android/view/VoucherCheckoutView.java

```diff
@@ -0,0 +1,111 @@
+package pl.rspective.voucherify.android.view;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+
+import pl.rspective.voucherify.android.client.R;
+import pl.rspective.voucherify.android.client.VoucherifyAndroidClient;
+import pl.rspective.voucherify.android.client.callback.VoucherifyCallback;
+import pl.rspective.voucherify.android.client.model.VoucherResponse;
+import retrofit.RetrofitError;
+
+/**
+ * VoucherCheckoutView is a UI component that makes it easier to validate discount codes in
+ * <a href="http://voucherify.io">Voucherify</a>. The view consists of a text field to input
+ * voucher code and a button to send the code for validation. You can get the validation result by
+ * attaching an @{link OnValidatedListener}.
+ */
+public class VoucherCheckoutView extends LinearLayout {
```

Comment:
Right now one view hierarchy is redundant in my opinion. You could extends by RelativeLayout and in XML file where you define the view remove the root RelativeLayout and replace with 'merge'